routes/api/comments.js

```diff
@@ -0,0 +1,63 @@
+//2 rts
+const express = require('express');
+const {asyncHandler} = require('../utils');
+const { restoreUser } = require('../../auth');
+const createError = require('http-errors');
+const router = express.Router();
+const { sequelize, CommentLike } = require("../../db/models");
+
+/* Like a comment */
+router.post('/comments/:id/likes', restoreUser, asyncHandler(async(req, res) => {
+
+    if (!res.locals.authenticated || res.locals.user.id !== like.userId) {
+        /* Return error 401, if
+               user is not authenticated, or
+               user is not liker */
+        return next(createError(401)); 
+
+    } else if (res.locals.authenticated ) {
+        
+        /* Create comment like */
+        const likedComment = await CommentLike.create({
+                userId: res.locals.user.id,
+                commentId: req.params.id
+            })
+        }
+
+        /* Set status 201, and
+           respond with json, likedComment */
+        res.status(201).json(likedComment);
+    }));
+
+
+/* Unlike a comment */
+router.delete('/comments/likes/:id', restoreUser, asyncHandler(async(req, res) => {
+
+        /* Initialize like object */
+        const like = await CommentLike.findByPk(req.params.id);
+
+        /* Return error 404, if 
+           there is no like object */
+        if (!like) {
+            return next(createError(404)); 
```

Comment:
The code does not have access to a `next` in this scope

Reply:
wait i had this at res.status(404) before these recent changes

Reply:
@kenjulian 

In an earlier comment, Justin wrote:

"res.status just sets the header of the response. Sending it using res.statusCode(404) is an option, but I think it's best we utilize error handlers here. We'll also need to make sure that the rest of the code does not attempt to execute, so be sure to return in the if statement.

"To utilize error handlers, use next with an error similar to when res.locals.authenticated is false."

This code reflects my previous fix, but I will continue to work on it.

Reply:
`next` can now be accessed.